### PR TITLE
[ModuleMonitor] Implement Timescalebd

### PR DESCRIPTION
**Pull request by alxtkr77**
### 📝 Description
  This PR implements TimescaleDB as a Time Series Database (TSDB) backend for MLRun's Model Monitoring system, providing a production-ready alternative to TDEngine and V3IO for storing and querying model
  monitoring metrics, predictions, and drift analysis results.

  TimescaleDB offers enterprise-grade reliability, SQL compatibility, and native time-series optimizations through continuous aggregates and hypertables.

  ---

  ### 🛠️ Changes Made

  **Core Implementation:**
  - Added `TimescaleDBConnector` with full TSDB interface implementation
  - Implemented `TimescaleDBConnection` with connection pooling and retry logic for deadlock handling
  - Created `TimescaleDBOperationsManager` for schema creation, table management, and data operations
  - Implemented `TimescaleDBStreamProcessor` for streaming data ingestion via Storey targets

  **Query System:**
  - Implemented modular query handler with three specialized mixins:
    - `TimescaleDBMetricsQueries` - Metrics operations (read, write, list)
    - `TimescaleDBPredictionsQueries` - Predictions and latency queries
    - `TimescaleDBResultsQueries` - Drift analysis and application results
  - Created `TimescaleDBQueryBuilder` utility for SQL generation with fallback patterns
  - Implemented `TimescaleDBDataFrameProcessor` for result transformation

  **Pre-Aggregation System:**
  - Implemented continuous aggregates for performance optimization at multiple intervals
  - Created `TimescaleDBPreAggregateManager` with automatic fallback to raw queries
  - Added intelligent interval selection based on query time range

  **Schema & Configuration:**
  - Defined TimescaleDB-specific table schemas with hypertable configuration
  - Added `DatastoreProfilePostgreSQL` for connection string management
  - Updated stream processing graph to support TimescaleDB targets
  - Added PostgreSQL/TimescaleDB dependency (`asyncpg~=0.30.0`)

  **Testing:**
  - Comprehensive test suite with 93 tests covering:
    - Connection management and retry logic (10 tests)
    - Operations (table creation, deletion, writes) (19 tests)
    - Query operations across all query types (23 tests)
    - Pre-aggregate functionality (26 tests)
    - DataFrame processing (3 tests)
    - Cross-query integration (3 tests)
    - Edge cases and validation (9 tests)
  - Test fixtures for TimescaleDB setup with Docker support
  - Integration with existing model monitoring system tests

  ---

  ### ✅ Checklist
  - [x] I updated the documentation (if applicable)
  - [x] I have tested the changes in this PR

  ---

  ### 🧪 Testing

  **Unit Tests:**
  - 93 TimescaleDB-specific tests (100% pass rate)
  - Connection retry and deadlock handling
  - Pre-aggregate with fallback patterns
  - Query operations for metrics, predictions, and results
  - DataFrame transformation and column mapping

  **Integration Tests:**
  - End-to-end model monitoring application flow
  - Streaming data ingestion via Storey
  - Multi-interval continuous aggregate queries
  - Cross-TSDB compatibility (TDEngine, V3IO, TimescaleDB)

  ### 🚨 Breaking Changes?

  - [x] No

  No breaking changes. TimescaleDB is added as an optional TSDB backend alongside existing TDEngine and V3IO implementations. Existing model monitoring deployments are unaffected.

  ---

  ### 🔍️ Additional Notes
Jira story: ML-11375

  **Architecture Highlights:**
  - **Modular Design**: Query handlers use mixin pattern for clean separation of concerns
  - **Performance**: Continuous aggregates provide 10-100x speedup for large time ranges
  - **Reliability**: Built-in deadlock detection and retry with exponential backoff
  - **SQL Compatibility**: Standard PostgreSQL interface for easy debugging and integration
  - **Connection Pooling**: Efficient connection management with configurable pool sizes

  **File Organization:**
  mlrun/model_monitoring/db/tsdb/timescaledb/
  ├── timescaledb_connector.py          # Main TSDB interface
  ├── timescaledb_connection.py         # Connection pooling & retry
  ├── timescaledb_operations.py         # Schema & table operations
  ├── timescaledb_schema.py             # Table definitions
  ├── timescaledb_stream.py             # Streaming processor
  ├── timescaledb_stream_graph_steps.py # Graph integration
  ├── writer_graph_steps.py             # Write operations
  ├── queries/
  │   ├── timescaledb_metrics_queries.py
  │   ├── timescaledb_predictions_queries.py
  │   └── timescaledb_results_queries.py
  └── utils/
      ├── timescaledb_query_builder.py
      ├── timescaledb_dataframe_processor.py
      └── (pre_aggregate logic integrated)

### CODE REVIEW

**Review by assaf758 — APPROVED** [review]
Following review in the feature-branch https://github.com/mlrun/mlrun/pull/8562.

File: extras-requirements.txt [hunk 1]
```diff
@@ -51,4 +51,6 @@ distributed~=2023.12.1; python_version < "3.11"
 dask==2024.8; python_version >= "3.11"
 distributed==2024.8; python_version >= "3.11"
 taos-ws-py==0.3.2
+# TimescaleDB dependencies - optional for model monitoring
+psycopg[binary,pool]~=3.2
```
Comment (liranbg) [inline extras-requirements.txt:55]: note you didnt run `lock` so no image current use it
